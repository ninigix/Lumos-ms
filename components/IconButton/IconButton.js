import React from "react";
import { Image, View } from "react-native";

import styles from "./IconButton.style";

const IconButton = ({ type, wasSelected }) => {
  const helper = {
    clock: require("../../assets/icons/time.png"),
    kitchen: require("../../assets/icons/kitchen.png"),
    bathroom: require("../../assets/icons/bathroom.png"),
    hall: require("../../assets/icons/hall.png"),
    bedroom1: require("../../assets/icons/bedroom.png"),
    bedroom2: require("../../assets/icons/bedroom.png"),
    lightbulb: require("../../assets/icons/lightbulb.png")
  };

  return (
    <View
      style={{
        ...styles.iconButton,
        backgroundColor: wasSelected ? "#ffbf00" : "#EBEBE4"
      }}
    >
      <Image style={styles.icon} source={helper[type]} />
    </View>
  );
};

export default IconButton;
