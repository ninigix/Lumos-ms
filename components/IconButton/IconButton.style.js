import { StyleSheet } from "react-native";

export default StyleSheet.create({
  iconButton: {
    height: 40,
    width: 40,
    borderRadius: 4,
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    margin: 20
  },
  icon: {
    width: 30,
    height: 30
  }
});
