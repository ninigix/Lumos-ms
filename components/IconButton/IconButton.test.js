import renderer from 'react-test-renderer';
import React from 'react';

import IconButton from './IconButton';

it('renders correctly', () => {
    const tree = renderer
        .create(
            <IconButton />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
