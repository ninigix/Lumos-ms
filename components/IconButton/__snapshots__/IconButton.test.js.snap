// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EBEBE4",
      "borderRadius": 4,
      "display": "flex",
      "height": 40,
      "justifyContent": "center",
      "margin": 20,
      "width": 40,
    }
  }
>
  <Image
    style={
      Object {
        "height": 30,
        "width": 30,
      }
    }
  />
</View>
`;
